### ISSUE TITLE
Support republishing optional ObservableObjects

### ISSUE THREAD

**Issue by FlorianHubl** [adam-zethraeus/Republished#4]
It doesn't work with optionals.

```swift
@Republished var a: A?
```

Error: Generic class 'Republished' requires that 'A?' conform to 'ObservableObject'

**Comment by adam-zethraeus** [comment]
This request is very similar to #3 — it's doable with some generics work, but doesn't behave like SwiftUI.

SwiftUI doesn't allow publishing from optional ObservableObjects.

```swift
public struct ContentView: View {

  // this is fine
  @StateObject var test = TestObject()

  // Generic struct 'StateObject' requires that 'TestObject?' conform to 'ObservableObject'
  @StateObject var test2: TestObject?

    public var body: some View {
      VStack {
        Text("hi")
      }
    }
}
```

I'll consider implementing this! But in the meantime anyone should feel free to signal desire, or even put up a PR :).

**Comment by adam-zethraeus** [comment]
@FlorianHubl could you try targeting the experimental branch in https://github.com/adam-zethraeus/Republished/pull/5 and see if it works well for you?

**Comment by adam-zethraeus** [comment]
implemented in 1.1
